### PR TITLE
Feature/adding example for support module

### LINKED ISSUE #138: split support activity & fragment logic into separate module
that way we could offer a base  library without integrating android support library, which would reduce transitive dependency size. @bblackbelt what do you think?

### PR DESCRIPTION
**Pull request by mrsasha**
this should finish this feature - copying example from ABExample to ABExample-support and adapting ABExample to use the base library.

closes https://github.com/alter-ego/androidbound/issues/138 and https://github.com/alter-ego/androidbound/issues/144.

### CODE REVIEW

**Comment by bblackbelt** [comment]
@mrsasha it needs rebase

**Comment by mrsasha** [comment]
@bblackbelt rebased

File: AndroidBoundExample-support/src/main/java/solutions/alterego/androidbound/example/support/MainActivity.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    protected void onSaveInstanceState(Bundle outState) {
+        super.onSaveInstanceState(outState);
+        mViewModel.onSaveInstanceState(outState);
+    }
+
+    @Override
+    protected void onDestroy() {
+        super.onDestroy();
+        mViewModel.dispose();
+        mViewBinder.disposeOf(this);
```
Comment (bblackbelt) [inline AndroidBoundExample-support/src/main/java/solutions/alterego/androidbound/example/support/MainActivity.java:72]: why do you need this line?

File: AndroidBoundExample-support/src/main/java/solutions/alterego/androidbound/example/support/nestedrvs/viewmodel/MainNestedViewModel.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                return true;
+            }
+            if (!(o instanceof RecyclerViewItem)) {
+                return false;
+            }
+            final RecyclerViewItem other = (RecyclerViewItem) o;
+            if (!other.canEqual((Object) this)) {
+                return false;
+            }
+            final Object this$mName = this.getName();
```
Comment (bblackbelt) [inline AndroidBoundExample-support/src/main/java/solutions/alterego/androidbound/example/support/nestedrvs/viewmodel/MainNestedViewModel.java:55]: why the $?

**Review by bblackbelt — CHANGES_REQUESTED** [review]

File: AndroidBoundExample-support/src/main/java/solutions/alterego/androidbound/example/support/nestedrvs/viewmodel/MainNestedViewModel.java (hunk 2 above)
Comment (mrsasha) [reply AndroidBoundExample-support/src/main/java/solutions/alterego/androidbound/example/support/nestedrvs/viewmodel/MainNestedViewModel.java:55]: hm, it was done automatically when delombokizing by the lombok plugin. :(

File: AndroidBoundExample-support/src/main/java/solutions/alterego/androidbound/example/support/MainActivity.java (hunk 1 above)
Comment (mrsasha) [reply AndroidBoundExample-support/src/main/java/solutions/alterego/androidbound/example/support/MainActivity.java:72]: it makes sure all the references are released. most of them shouldn't be a problem, but I don't know about the `FontManager` that has typefaces, so this is a "better safe than sorry" thing.

File: AndroidBoundExample-support/src/main/java/solutions/alterego/androidbound/example/support/nestedrvs/viewmodel/MainNestedViewModel.java (hunk 2 above)
Comment (bblackbelt) [reply AndroidBoundExample-support/src/main/java/solutions/alterego/androidbound/example/support/nestedrvs/viewmodel/MainNestedViewModel.java:55]: do you want to change it ?

**Pull request merged**